ansible/roles_ocp_workloads/ocp4_workload_codeserver/tasks/workload.yml

```diff
@@ -25,55 +21,113 @@
+
+  - name: Save code server information
     agnosticd_user_info:
-      msg: "{{ item }}"
-    loop:
-    - ""
-    - "Code server has been setup for your user:"
-    - "  URL: https://{{ ocp4_workload_codeserver_name }}-{{ ocp4_workload_codeserver_namespace
-     }}.{{ _ocp4_workload_codeserver_wildcard_domain }}"
-    - "  Password: {{ ocp4_workload_codeserver_password }}"
+      data:
+        codeserver_url: >-
+          https://{{ ocp4_workload_codeserver_name }}-{{ ocp4_workload_codeserver_namespace
+          }}.{{ _ocp4_workload_codeserver_wildcard_domain }}
+        codeserver_password: "{{ ocp4_workload_codeserver_password }}"
 
 - name: Multi User installation
   when: ocp4_workload_codeserver_multi_user_install | bool
   block:
-  - name: Install code server
-    ansible.builtin.include_tasks: install_multi_user.yml
-    loop: "{{ range(1, ocp4_workload_codeserver_multi_user_num_users | int + 1) | list }}"
+  # Build dictionary of repositories
+  - name: Print repos to be added
+    when: ocp4_workload_codeserver_repos | length > 0
+    debug:
+      msg: |-
+        {{ _ocp4_workload_codeserver_gitea_protocol
+        }}://{{'{{ user }}'}}:{{
+        ((ocp4_workload_codeserver_repos_password | default('') ~ '@') if ocp4_workload_codeserver_repos_password | default('') | length > 0 else '')
+        }}{{ _ocp4_workload_codeserver_gitea_hostname }}/{{'{{ user }}'}}/{{ repo.name }}
+    loop: "{{ ocp4_workload_codeserver_repos }}"
     loop_control:
-      loop_var: n
-      label: "{{ ocp4_workload_codeserver_multi_user_username_base }}{{ n }}"
+      loop_var: repo
+      label: "{{ repo.name }}"
+
+  - name: Add repos to dictionary
+    when: ocp4_workload_codeserver_repos | length > 0
+    ansible.builtin.set_fact:
+      _ocp4_workload_codeserver_repos: |-
+        {{ _ocp4_workload_codeserver_repos + [
+        { 'name': repo.name,
+          'url': _ocp4_workload_codeserver_gitea_protocol ~ '://' ~
+                  '{{ user }}:' ~
+                  ((ocp4_workload_codeserver_repos_password | default("") ~ "@") if ocp4_workload_codeserver_repos_password | default("") | length > 0
+                  else "") ~
+                  _ocp4_workload_codeserver_gitea_hostname ~ '/' ~ '{{ user }}' ~ "/" ~ repo.name,
+          'branch': repo.branch } ]
+        }}
+    loop: "{{ ocp4_workload_codeserver_repos }}"
+    loop_control:
+      loop_var: repo
+      label: "{{ repo.name }}"
+
+  - name: Print repos dictionary (multi user)
+    debug:
+      msg: "{{ _ocp4_workload_codeserver_repos }}"
+
+  # - name: Debug template
+  #   template:
+  #     src: applicationset.yaml.j2
+  #     dest: /home/ec2-user/appset-codeserver.yaml
+  #     owner: ec2-user
+  #     mode: 0664
+
+  - name: Install code server (multi user)
+    kubernetes.core.k8s:
+      state: present
+      definition: "{{ lookup('template', 'applicationset.yaml.j2') }}"
 
   - name: Save code server information for each user
     agnosticd_user_info:
```

Comment:
JK wants switches on agonsticd_user_info to allow it to be shut off.

Reply:
Happy to add - once I understand what the point of that would be. I mean all information that we provide is necessary...